# netsurf: image-handlers
# 5 files, 7 changed lines
#
# Regenerate: tools/regen-patches.sh && tools/split-patches.py
#
diff -urN -x .git -x 'build-*' -x test -x frontends -x '*.o' -x '*.a' -x '*.d' -x '*.pc' -x build upstream/netsurf/content/handlers/image/bmp.c browser/netsurf/content/handlers/image/bmp.c
--- upstream/netsurf/content/handlers/image/bmp.c	2026-09-20 23:29:00.060762334 +0100
+++ browser/netsurf/content/handlers/image/bmp.c	2026-09-20 23:19:20.745477382 +0100
@@ -200,6 +200,7 @@
 		flags |= BITMAPF_REPEAT_X;
 	if (data->repeat_y)
 		flags |= BITMAPF_REPEAT_Y;
+	flags |= BITMAPF_BLEND_MODE(data->background_blend_mode);
 
 	return (ctx->plot->bitmap(ctx,
 				  bmp->bitmap,
diff -urN -x .git -x 'build-*' -x test -x frontends -x '*.o' -x '*.a' -x '*.d' -x '*.pc' -x build upstream/netsurf/content/handlers/image/image.c browser/netsurf/content/handlers/image/image.c
--- upstream/netsurf/content/handlers/image/image.c	2026-09-20 23:29:00.060762334 +0100
+++ browser/netsurf/content/handlers/image/image.c	2026-09-20 23:19:20.746264896 +0100
@@ -168,6 +168,7 @@
 		flags |= BITMAPF_REPEAT_X;
 	if (data->repeat_y)
 		flags |= BITMAPF_REPEAT_Y;
+	flags |= BITMAPF_BLEND_MODE(data->background_blend_mode);
 
 	return (ctx->plot->bitmap(ctx,
 				  bitmap,
diff -urN -x .git -x 'build-*' -x test -x frontends -x '*.o' -x '*.a' -x '*.d' -x '*.pc' -x build upstream/netsurf/content/handlers/image/nssprite.c browser/netsurf/content/handlers/image/nssprite.c
--- upstream/netsurf/content/handlers/image/nssprite.c	2026-09-20 23:29:00.060762334 +0100
+++ browser/netsurf/content/handlers/image/nssprite.c	2026-09-20 23:19:20.747424107 +0100
@@ -200,6 +200,7 @@
 	if (data->repeat_y) {
 		flags |= BITMAPF_REPEAT_Y;
 	}
+	flags |= BITMAPF_BLEND_MODE(data->background_blend_mode);
 
 	return (ctx->plot->bitmap(ctx,
 				  nssprite->bitmap,
diff -urN -x .git -x 'build-*' -x test -x frontends -x '*.o' -x '*.a' -x '*.d' -x '*.pc' -x build upstream/netsurf/content/handlers/image/rsvg.c browser/netsurf/content/handlers/image/rsvg.c
--- upstream/netsurf/content/handlers/image/rsvg.c	2026-09-20 23:29:00.060762334 +0100
+++ browser/netsurf/content/handlers/image/rsvg.c	2026-09-20 23:19:20.747917369 +0100
@@ -204,6 +204,7 @@
 		flags |= BITMAPF_REPEAT_X;
 	if (data->repeat_y)
 		flags |= BITMAPF_REPEAT_Y;
+	flags |= BITMAPF_BLEND_MODE(data->background_blend_mode);
 
 	return (ctx->plot->bitmap(ctx,
 				  rsvgcontent->bitmap,
@@ -314,4 +315,3 @@
 };
 
 CONTENT_FACTORY_REGISTER_TYPES(nsrsvg, rsvg_types, rsvg_content_handler);
-
diff -urN upstream/netsurf/content/handlers/image/video.h browser/netsurf/content/handlers/image/video.h
--- upstream/netsurf/content/handlers/image/video.h
+++ browser/netsurf/content/handlers/image/video.h
@@ -19,7 +19,7 @@
 #ifndef NETSURF_IMAGE_VIDEO_H_
 #define NETSURF_IMAGE_VIDEO_H_
 
-#include "utils/errors.h"
+#include "utils/errors.h"
 
 nserror nsvideo_init(void);
 
